src/Microsoft.DotNet.TemplateLocator/TemplateLocator.cs

```diff
@@ -5,68 +5,106 @@
-            if (!string.IsNullOrWhiteSpace(mockTemplateLocation))
-            {
-                _dotnetSdkTemplatesLocation = new DirectoryInfo(mockTemplateLocation);
-            }
         }
 
-        public IReadOnlyCollection<IOptionalSdkTemplatePackageInfo> GetDotnetSdkTemplatePackages(string sdkVersion)
+        // Test constructor
+        public TemplateLocator(Func<string, string> getEnvironmentVariable, VSSettings vsSettings)
+        {
+            _netCoreSdkResolver =
+                new Lazy<NETCoreSdkResolver>(() => new NETCoreSdkResolver(getEnvironmentVariable, vsSettings));
+        }
+
+        public IReadOnlyCollection<IOptionalSdkTemplatePackageInfo> GetDotnetSdkTemplatePackages(
+            string sdkVersion,
+            string dotnetRootPath)
         {
-            if (_dotnetSdkTemplatesLocation == null)
+            var sdkVersionParsed = Version.Parse(sdkVersion.Split('-')[0]);
+
+            bool FindSameVersionBand(DirectoryInfo manifestDirectory)
             {
-                return Array.Empty<IOptionalSdkTemplatePackageInfo>();
+                var directoryVersion = Version.Parse(manifestDirectory.Name.Split('-')[0]);
+                return directoryVersion.Major == sdkVersionParsed.Major &&
+                       directoryVersion.Minor == sdkVersionParsed.Minor &&
+                       (directoryVersion.Revision / 100) == (sdkVersionParsed.Revision / 100);
             }
 
-            IEnumerable<LocalPackageInfo> packages = LocalFolderUtility
-                            .GetPackagesV2(_dotnetSdkTemplatesLocation.FullName, new NullLogger());
+            var bondManifestDirectory = new DirectoryInfo(dotnetRootPath).GetDirectories().Where(FindSameVersionBand)
+                .OrderByDescending(d => Version.Parse(d.Name.Split('-')[0])).FirstOrDefault();
 
-            if (packages == null)
+            if (bondManifestDirectory == null)
             {
                 return Array.Empty<IOptionalSdkTemplatePackageInfo>();
             }
-            else
+
+            var manifestContent =
+                WorkloadManifest.LoadFromFolder(bondManifestDirectory.GetDirectories().Single().FullName);
+            var dotnetSdkTemplatePackages = new List<IOptionalSdkTemplatePackageInfo>();
+            foreach (var pack in manifestContent.SdkPackDetail)
             {
-                return packages
-                    .Select(l => new OptionalSdkTemplatePackageInfo(l)).ToArray();
+                if (pack.Value.kind.Equals("Template", StringComparison.OrdinalIgnoreCase))
+                {
+                    var optionalSdkTemplatePackageInfo = new OptionalSdkTemplatePackageInfo(pack.Key,
+                        pack.Value.version,
+                        Path.Combine(dotnetRootPath, "template-packs",
+                            pack.Key.ToLower() + "." + pack.Value.version.ToLower() + ".nupkg"));
```

Comment:
minor, inconsistent spacing of ```OptionalSdkTemplatePackageInfo``` .ctor params

Reply:
it is formatted by IDE. Which part is inconsistent?